scripts/reportInstalls.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { promisify } from 'util';
+import { exec } from 'child_process';
+import * as appInsights from 'applicationinsights';
+import { DEFAULT_AIKEY } from '../packages/salesforcedx-utils-vscode/src/constants';
+
+const promisifiedExec = promisify(exec);
+
+interface InstallTelemetryData {
+  // To align with the properties property for appInsightsClient.trackEvent
+  [key: string]: string;
+  install: string;
+  ratingcount: string;
+  marketplace: string;
+  'common.extname': string;
+}
+
+// These properties match the values provided in the statistics array from the
+// vsce cli output
+interface InstallTelemetryDataMS extends InstallTelemetryData {
+  averagerating: string;
+  trendingdaily: string;
+  trendingmonthly: string;
+  trendingweekly: string;
+  updateCount: string;
+  weightedRating: string;
+  downloadCount: string;
+}
+
+type TelemetryData = InstallTelemetryData | InstallTelemetryDataMS;
```

Comment:
Tried to ensure we send the data we think we're sending and if we change it the type system will let us know.